### PR TITLE
issue-1363: Allow null in "selectedKey" to avoid typing problems, doc…

### LINKED ISSUE #1363: useSingleSelectListState should support having no selected key
# 🙋 Feature Request

Right now, `SingleSelectListState.selectedKey` is a required field, and `useSingleSelectListState` interprets an undefined selectedKey as "uncontrolled" (vs controlled with an undefined value).  `useSingleSelectListState` also calls `useListState` with `disallowEmptySelection`.

Workaround right now is to use `null` as the selectedKey, but there seems to be no official way to clear a selection.

## 🤔 Expected Behavior

Can have a SingleSelectListState with no selectedKey

## 😯 Current Behavior

SingleSelectListState.selectedKey, once set, cannot be unset

## 💁 Possible Solution

There is some ambiguity here on how `useControlledState` interprets value and defaultValue; `undefined` could both mean "no prop passed in" or "undefined passed in".  Could switch to using, say, `null`, or some other defined sentinel / symbol to indicate "no prop passed in" instead:

```
function keyOrNone(props, key) {
  return key in props ? props[key] : NONE;
}

function useSingleSelectListaState() {  
  useControlledState(keyOrNone(props, "selectedKey"), keyOrNone(props, "defaultSelectedKey"), ...);
}
```

## 🔦 Context

Trying to implement a Select / Picker that can be deselected.

## 💻 Examples

In this example: https://codesandbox.io/s/clever-dirac-vtrd6?file=/src/App.tsx

Using Spectrum Picker, selecting an option and then selecting the "Unset" option (which sets the selectedKey to `undefined`) results in unexpected behavior (Picker still shows a selected value).

## 🧢 Your Company/Team

Plasmic

### PR DESCRIPTION
**Pull request by khoad**
…ument using "null" to clear key selection

Closes #1363

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

* Open story book and navigate to Picker -> 'can be deselected'
* Open docs and go to 'Deselection' section in 'Selection' section

## 🧢 Your Project:

Adobe

### CODE REVIEW

File: packages/@react-spectrum/picker/stories/Picker.stories.tsx [hunk 1]
```diff
@@ -63,6 +63,19 @@ storiesOf('Picker', module)
       </Picker>
     )
   )
+  .add(
+    'can be deselected',
+    () => {
+      const [selected, setSelected] = React.useState<React.Key>();
+      return (
+        <Picker label="Test" selectedKey={selected} onSelectionChange={val => setSelected(val === selected ? null : val)}>
```
Comment (khoad) [inline packages/@react-spectrum/picker/stories/Picker.stories.tsx:71]: I realize this looks different than the rest of the stories, but couldn't find a better way to do this. If you have suggestions let me know. Thanks.

**Review by snowystinger — DISMISSED** [review]
Looks good, we can also publish the docs because this will technically already work, people may just need to ts-ignore for a little bit until the types are updated

**Review by reidbarber — APPROVED** [review]
Looks great! Thanks for adding.

**Comment by dannify** [comment]
Removed the docs update because this is more of a workaround, the rest looks good.

**Review by dannify — DISMISSED** [review]

File: packages/@react-spectrum/picker/docs/Picker.mdx [hunk 2]
```diff
@@ -151,6 +151,7 @@ function Example() {
   );
 }
 ```
+
```
Comment (dannify) [inline packages/@react-spectrum/picker/docs/Picker.mdx:154]:
```suggestion
```

**Review by devongovett — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Pull request merged**